src/AspNetCoreModuleV2/AspNetCore/src/applicationinfo.cpp

```diff
@@ -222,8 +223,16 @@ APPLICATION_INFO::EnsureApplicationCreated(
                 FINISHED(HRESULT_FROM_WIN32(ERROR_INVALID_FUNCTION));
             }
 
-            FINISHED_IF_FAILED(m_pfnAspNetCoreCreateApplication(m_pServer, pHttpContext->GetApplication(), &pApplication));
-            pApplication->SetParameter(L"InProcessExeLocation", struExeLocation.QueryStr());
+            std::array<APPLICATION_PARAMETER, 1> parameters {
+                {"InProcessExeLocation", struExeLocation.QueryStr()}
```

Comment:
This won't copy the exeLocation. I don't like that it's up to the application to copy all pointers in the application_parameter array.

Reply:
I would prefer not to manage lifetime for this array, especially if app doesn't need it.

Reply:
And it was the same in SetParameter before.

Reply:
IMO if we pass something from the shim to the request handler, ownership should transfer completely. If someone stores the application parameter array and expects the VOID* not to be freed, then we should copy the values.

Reply:
How do you know how to free void* correctly without knowing data type?

Reply:
We don't, but it would avoid bugs where someone doesn't copy. Like here: https://github.com/aspnet/IISIntegration/pull/998/files#diff-4653e8c4233ce04437f43552addb24f3R42 ; instead of calling Copy, you just assign the string somewhere, it would AV.

Reply:
How do you free the memory after application would need to get recycled?